main.rb

```diff
@@ -0,0 +1,161 @@
+      i += 1
+    end
+    self
+  end
+
+  # my_select method
+  def my_select
+    return to_enum unless block_given?
+
+    arr = []
+    to_a.my_each { |i| arr << i if yield i }
+    arr
+  end
+
+  # my_all method
+  def my_all?(args = nil)
+    if block_given?
+      my_each { |i| return false if yield(i) == false }
+      return true
+    elsif args.nil?
+      my_each { |i| return false if i.nil? || i == false }
+    elsif !args.nil? && (args.is_a? Class)
+      my_each { |i| return false unless i.is_a?(args) }
+    elsif !args.nil? && args.class == Regexp
+      my_each { |i| return false unless args.match(i) }
+    else
+      my_each { |i| return false if i != args }
+    end
+    true
+  end
+
+  # my_any method
+  def my_any?(args = nil)
+    if block_given?
+      to_a.my_each { |i| return true if yield(i) }
+      return false
+    elsif args.nil?
+      to_a.my_each { |i| return true if i }
+    elsif !args.nil? && (args.is_a? Class)
+      to_a.my_each { |i| return true if i.is_a?(args) }
+    elsif !args.nil? && args.class == Regexp
+      to_a.my_each { |i| return true if args.match(i) }
+    else
+      to_a.my_each { |i| return true if i == args }
+    end
+    false
+  end
+
+  # my_none method
+  def my_none?(arg = nil)
+    if block_given?
+      my_each { |i| return false if yield i }
+    elsif arg.nil?
+      my_each { |i| return false if i }
+    elsif arg.is_a?(Class)
+      my_each { |i| return false if i.class == arg }
+    elsif arg.class == Regexp
+      my_each { |i| return false if arg.match(i) }
+    else
+      my_each { |i| return false if i == arg }
+    end
+    true
+  end
+
+  # my_count method
+  def my_count(args = nil)
+    count = 0
+    if args.nil?
+      block_given? ? to_a.my_each { |i| count += 1 if yield(i) } : count = to_a.length
+    else
+      to_a.my_each { |i| count += 1 if args == i }
+    end
+    count
+  end
+
+  # my_map method
+  def my_map(proc = nil)
+    return to_enum unless block_given? || !proc.nil?
+
+    arr = []
+    if proc
+      my_each { |i| arr << proc.call(i) }
+    else
+      my_each { |i| arr << yield(i) }
+    end
+    arr
+  end
+
+  # my_inject method
+  def my_inject(*acc)
+    result = 0
+    arr = to_a
+    if acc.size == 1
+      if acc[0].is_a?(Symbol)
+        case acc[0]
+        when :+
+          arr.my_each { |a| result += a }
+          result
+        when :-
+          arr.my_each { |a| result -= a }
+          result
+        when :*
+          result = 1
+          arr.my_each { |a| result *= a }
+          result
+        end
+      elsif acc[0].is_a?(Integer)
+        k = acc[0]
+        arr.my_each { |a| k = yield(k, a) }
+        k
+      end
+    elsif acc.size == 2
+      result = acc[0]
+      case acc[1]
+      when :+
+        arr.my_each { |a| result += a }
+        result
+      when :-
+        arr.my_each { |a| result -= a }
+        result
+      when :*
+        arr.my_each { |a| result *= a }
+        result
+      end
+    else
```

Comment:
- [ ] It is not working for division and subtraction. One option for the symbol case is to use the _".send"_. Check out, I think it would make your method shorter. Feel free to fix this however you choose, but those cases in the print screen need to work. 

![image](https://user-images.githubusercontent.com/61479648/93529383-33ae2600-f912-11ea-843f-bf977d7a4a91.png)